### PR TITLE
Fix Tensor.mean() silently dropping gradients

### PR DESCRIPTION
**Pull request by Shashank-Tripathi-07**
## What's broken

`autograd.py` patches `Tensor.sum = sum_op`, a version of `sum()` that builds a `SumBackward` node and correctly propagates `requires_grad`. There's no equivalent for `mean()` anywhere in the curriculum, so it silently keeps running module 01's original plain implementation: `np.mean(...)` wrapped in a fresh `Tensor(result)`, with no gradient tracking at all.

```python
t = Tensor([1.0, 2.0, 3.0], requires_grad=True)
out = (t * t).mean()
out.requires_grad  # False
out.backward()
t.grad  # None -- no error, no warning, just silently wrong
```

`sum()` on the identical input correctly returns `requires_grad=True` and backprops. Nothing inside TrenTorch's own losses/layers currently calls `.mean()` (confirmed via a full grep of the exported package), so no internal code path breaks today, but any student who reaches for the natural `.mean()` call in their own training loop, e.g. averaging a batch loss, gets silently zero gradients with no error and no way to know their own code isn't the problem.

## The fix

`mean_op` is built from the already-tracked `sum()` and division rather than a new backward `Function` from scratch, since both are already verified correct elsewhere. Handles `axis=None`, an int axis, and a tuple axis the same way `np.mean` does. Wired in as `Tensor.mean = mean_op` right alongside the existing `Tensor.sum = sum_op` line.

## Verification

- Reproduced the bug directly, confirmed the fix resolves it with the exact expected analytical value (`2x/n`)
- 20-trial finite-difference check against random inputs, all matched within 5e-3 tolerance
- Full local suite: 897 passed / 2 failed, identical to the pre-fix baseline. Both remaining failures are pre-existing and unrelated (this dev machine's local `pip` absence, and a known test-isolation gap already tracked in `docs/testing-strategy.md`)
- `ruff check` / `ruff format --check` clean
- The edit to `data/src/06_autograd/06_autograd.py` is confined entirely inside the existing, already-fully-implemented `enable_autograd()` function (no `# %%` cell markers touched), verified via diff against HEAD before and after

### CODE REVIEW

**Comment by Shashank-Tripathi-07** [comment]
Closing to reopen from the correctly-named branch per the new branch naming convention (docs/CONTRIBUTING.md: <github-username>/<feature-name>). Same commit, no content change.
